Git Streak Tracker/SettingsView.swift

```diff
@@ -10,29 +10,94 @@ import SwiftUI
 import WidgetKit
 import SwiftUIFontIcon
 
+let debouncer = Debouncer(delay: 0.5)
+
 struct SettingsView: View {
     @EnvironmentObject private var userStore: UserStore
+    @EnvironmentObject private var viewStore: ViewStore
     @State private var inputValue: String = ""
     @State private var isEditing = false
     
-    let debouncer = Debouncer(delay: 0.5)
-    
+    // Actions
     func loadInputField() {
         inputValue = userStore.username
     }
     
-    func storeUsername() {
-        userStore.username = inputValue // everytime username changes, the contributions are requested
+    func handleInputChanged() {
+        inputValue = inputValue.trimmingCharacters(in: .whitespacesAndNewlines)
+        userStore.error = !isValidGHUsername(uname: inputValue) && inputValue != ""
+        debouncer.renewInterval {
```

Comment:
We shouldn't trigger this `renewInterval` call in the case that the username is invalid.

Reply:
just [updated the debounce function](https://github.com/gibsonbailey/git-streak-tracker/pull/30/commits/7ea1efaeeb7b299a2a258e8a9c5e6b9954d460a7) to skip the request following the same logic, I think we still wanna call the debounce so it cancels the last potentially valid call.

Reply:
Thanks for the clarification. That's right.